apps/astarte_trigger_engine/lib/astarte_trigger_engine/events_consumer.ex

```diff
@@ -248,27 +245,38 @@ defmodule Astarte.TriggerEngine.EventsConsumer do
   end
 
   defp retrieve_trigger_configuration(realm_name, trigger_id) do
-    query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(
-        "SELECT value FROM kv_store WHERE group='triggers' AND key=:trigger_id;"
-      )
-      |> DatabaseQuery.put(:trigger_id, trigger_id)
+    statement =
+      "SELECT value FROM #{realm_name}.kv_store WHERE group='triggers' AND key=:trigger_id;"
+
+    params = %{"trigger_id" => trigger_id}
 
-    with {:ok, client} <- Database.connect(realm: realm_name),
-         {:ok, result} <- DatabaseQuery.call(client, query),
-         [value: trigger_data] <- DatabaseResult.head(result),
+    with :ok <- check_realm_name_ok(realm_name),
+         {:ok, prepared} <- Xandra.Cluster.prepare(:xandra, statement),
+         {:ok, result} <- Xandra.Cluster.execute(:xandra, prepared, params),
+         %{"value" => trigger_data} <-
+           Enum.at(result, 0, {:error, :invalid_query_result, result}),
          trigger <- Trigger.decode(trigger_data),
          {:ok, action} <- Jason.decode(trigger.action) do
       {:ok, action}
     else
-      {:error, :database_connection_error} ->
-        Logger.warn("Database connection error.")
+      {:error, %Xandra.ConnectionError{action: action, reason: reason}} ->
```

Comment:
The `{:error, %Xandra.Error{}}` case is not handled. It would be good to keep database and logic error (e.g. trigger not found) distinct